### PR TITLE
Test Fixes

### PR DESCRIPTION
**Pull request by cognusion**
A bunch of tests were failing, which was making my automatic cross-compiling a pain. This is a series of atomic fixes to clean them up.

### CODE REVIEW

**Comment by xiang90** [comment]
@cognusion gofmt checking fails.

File: raft/rafttest/network_test.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func TestNetworkDelay(t *testing.T) {
 
 	w := time.Duration(float64(sent)*delayrate/2) * delay
 	// there are pretty overhead in the send call, since it genarete random numbers.
-	if total < w+10*delay {
+	if total < w {
```
Comment (yichengq) [inline raft/rafttest/network_test.go]: I think this is correct and the above comment should be removed. The `rand.Int63n` takes 63ns/op.
/cc @xiang90

**Comment by yichengq** [comment]
@cognusion Could you follow https://github.com/coreos/etcd/blob/master/CONTRIBUTING.md#format-of-the-commit-message to give commit title? It helps commit history much readable.

File: raft/rafttest/network_test.go (hunk 1 above)
Comment (cognusion) [inline raft/rafttest/network_test.go]: On my laptop, with golang 1.5, it's consistently taking 57-59

**Comment by xiang90** [comment]
@cognusion Kindly ping on this one.

**Comment by cognusion** [comment]
Apologies, I hope to get to this tonight.

**Comment by cognusion** [comment]
This pull abandoned in favor of #3383 . Apologies for not reviewing the formatting guidelines initially.

**Pull request closed without merging**